hfinger/hfinger_core.py

```diff
@@ -98,132 +93,139 @@ def get_hdr_order(request_split):
                 hdr_coded = HDRL[hdr_lower]
             else:
                 hdr_coded = "!" + HDRL[hdr_lower]
-        ret.append(hdr_coded)
-    return ",".join(ret)
+        return_list.append(hdr_coded)
+    return ",".join(return_list)
 
 
 def get_ua_value(hdr):
-    val = hdr.split(":")[1]
-    if val[0] == " ":
-        val = val[1:]
+    header_value = hdr.split(":")[1].lstrip(" ")
     name = HDRL["user-agent"]
-    ret = name + ":" + format(fnv1a_32(val.encode()), "x")
+    ret = name + ":" + format(fnv1a_32(header_value.encode()), "x")
     return ret
 
 
 def get_hdr_value(hdr, hdrname, hdr_value_table):
-    val = hdr.split(":")[1]
-    if val[0] == " ":
-        val = val[1:]
+    header_value = hdr.split(":")[1].lstrip(" ")
```

Comment:
this is using the same code as `get_ua_value`, maybe the header parsing logic should be contained in a helper function or the whole request should be parsed beforehand?

Reply:
As suggested, I've added a helper function in https://github.com/CERT-Polska/hfinger/pull/6/commits/15c370829998ec4330fbce97fe0a30d5b6c5cb3e